### ISSUE TITLE
Execute CLANG Memory Sanitizer

### ISSUE THREAD

**Issue by mancoast** [buddhi1980/mandelbulber2#321]
When saving very large images, there exists a memory leak.

**Comment by mancoast** [comment]
End Users observe a large waiting period to save the image.
When we enter the realm of giga-pixels, I observe more time saving than rendering!

**Comment by mancoast** [comment]
REF: http://gigapan.com/gigapans/199297
This 16k image took ~300seconds to render, but over 10 minutes to save?

**Comment by buddhi1980** [comment]
Save in TIFF or JPG - will be faster. PNG compression is quite slow and single threaded.

**Comment by buddhi1980** [comment]
About memory leak, can you point where it is? In which format? Which image settings? I cannot observe any memory leaks.

About image saving times (just tested):
16000x16000 image:
JPG - 3s
PNG - 40s
PNG16 - 67s
TIFF - 25s
TIFF16 - 44s

**Comment by mancoast** [comment]
REF: http://www.gigapan.com/gigapans/199307
I pasted some more info below.
What I have noticed:
A1.  PID: 13496, time: 3912.223, Debug: OpenCl Rendering time [s] 2724.35
B2.  PID: 13496, time: 6430.333, Debug: Rendering time [s] 5242.87

I was saving the image after A1, but before B2.
Essentially, its possible to initiate the save after "Debug: OpenCl Rendering" but before the "Debug: Rendering time"
This applies to all of the formats.
The suggestion to use JPG/TIFF improves the save time.
However, its still possible to trigger unexpected behavior described above.
